templates/todo/api/java/src/main/java/com/microsoft/azure/simpletodo/configuration/AzureDeveloperCliCredential.java

```diff
@@ -0,0 +1,220 @@
+        StringBuilder azCommand = new StringBuilder("azd auth token --output json --scope ");
+
+        var scopes = String.join(" --scope ", request.getScopes());
+
+        try {
+            ScopeUtil.validateScope(scopes);
+        } catch (IllegalArgumentException ex) {
+            return Mono.error(LOGGER.logExceptionAsError(ex));
+        }
+
+        azCommand.append(scopes);
+
+        AccessToken token;
+        try {
+            String starter;
+            String switcher;
+            if (isWindowsPlatform()) {
+                starter = WINDOWS_STARTER;
+                switcher = WINDOWS_SWITCHER;
+            } else {
+                starter = LINUX_MAC_STARTER;
+                switcher = LINUX_MAC_SWITCHER;
+            }
+
+            ProcessBuilder builder = new ProcessBuilder(starter, switcher, azCommand.toString());
+
+            String workingDirectory = getSafeWorkingDirectory();
+            if (workingDirectory != null) {
+                builder.directory(new File(workingDirectory));
+            } else {
+                throw LOGGER.logExceptionAsError(
+                    new IllegalStateException(
+                        "A Safe Working directory could not be" + " found to execute Azure Developer CLI command from."
+                    )
+                );
+            }
+            builder.redirectErrorStream(true);
+            Process process = builder.start();
+
+            StringBuilder output = new StringBuilder();
+            try (
+                BufferedReader reader = new BufferedReader(
+                    new InputStreamReader(process.getInputStream(), StandardCharsets.UTF_8.name())
+                )
+            ) {
+                String line;
+                while (true) {
+                    line = reader.readLine();
+                    if (line == null) {
+                        break;
+                    }
+
+                    if (
+                        line.startsWith(WINDOWS_PROCESS_ERROR_MESSAGE) ||
+                        LINUX_MAC_PROCESS_ERROR_MESSAGE.matcher(line).matches()
+                    ) {
+                        throw LoggingUtil.logCredentialUnavailableException(
+                            LOGGER,
+                            identityClientOptions,
+                            new CredentialUnavailableException(
+                                "AzureDeveloperCliCredential authentication unavailable. Azure Developer CLI not installed." +
+                                "To mitigate this issue, please refer to the troubleshooting guidelines here at " +
+                                "https://aka.ms/azsdk/java/identity/azclicredential/troubleshoot"
+                            )
+                        );
+                    }
+                    output.append(line);
+                }
+            }
+            String processOutput = output.toString();
+
+            process.waitFor(10, TimeUnit.SECONDS);
```

Comment:
Do we need an explicit wait here? What if the above takes more than 10 seconds?

Reply:
This is coming from `following the same steps as the AzureCliCredential` .  Maybe they had to add this delay because the `az cli` is typically slow.
I have removed the delay and tested that it works for `azd` (much faster).  

Thank you